### PR TITLE
✨ v3 (feature): new Route method

### PR DESCRIPTION
**Pull request by wangjq4214**
related to #1829

### CODE REVIEW

**Comment by wangjq4214** [comment]
> @wangjq4214 Why you changed `Route` with `RouteInfo` ?

I think the method returned by this function is probably more commonly used, so a simpler name is needed. Changing the previous name is not quite right, I will change it back, do you have any idea about the name of this type?

File: register.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+// 🤖 Github Repository: https://github.com/gofiber/fiber
+// 📌 API Documentation: https://docs.gofiber.io
+
+package fiber
+
+// Register struct
+type Register struct {
+	app *App
+
+	Path string
```
Comment (efectn) [inline register.go:11]: This can be private i suppose.

File: register.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+//	     return c.Next()
+//	})
+//	app.Use("/api", func(c fiber.Ctx) error {
+//	     return c.Next()
+//	})
+//	app.Use("/api", handler, func(c fiber.Ctx) error {
+//	     return c.Next()
+//	})
+//
+// This method will match all HTTP verbs: GET, POST, PUT, HEAD etc...
```
Comment (efectn) [inline register.go:27]: Needs to update comments

**Review by efectn — CHANGES_REQUESTED** [review]

**Comment by efectn** [comment]
Router interface is not compatible for this stuff. It has path param in methods

**Comment by balcieren** [comment]
> Router interface is not compatible for this stuff. It has path param in methods

That's true. You are right.

**Comment by balcieren** [comment]
I think we should change Route struct name

**Review by efectn — APPROVED** [review]

**Comment by balcieren** [comment]
Actually `RouteFunc` might be a good name to seperate with `Route`

**Comment by ReneWerner87** [comment]
@wangjq4214 thx for the PR

can you create some test with route which have a deeper level

so for example 3 level 

so we can be sure that there can be scaled infinitely

**Comment by wangjq4214** [comment]
@efectn thx

File: register.go (hunk 2 above)
Comment (andrijana44) [reply register.go:49]: const chunks = [];

readable.on('readable', () => {
  let chunk;
  while (null !== (chunk = readable.read())) {
    chunks.push(chunk);
  }
});

readable.on('end', () => {
  const content = chunks.join('');
});

File: register.go (hunk 1 above)
Comment (andrijana44) [reply register.go]: const chunks = [];

readable.on('readable', () => {
  let chunk;
  while (null !== (chunk = readable.read())) {
    chunks.push(chunk);
  }
});

readable.on('end', () => {
  const content = chunks.join('');
});

**Pull request merged**